drivers/dpll/dpll_core.c

```diff
@@ -93,7 +93,7 @@ dpll_xa_ref_pin_add(struct xarray *xa_pins, struct dpll_pin *pin,
 			kfree(ref);
 			return ret;
 		}
-		refcount_set(&ref->refcount, 1);
+		refcount_set(&ref->refcount, 0);
```

Comment:
I think this set to 0 is not valid and produced:

[  420.471602] ------------[ cut here ]------------
[  420.471613] refcount_t: addition on 0; use-after-free.
[  420.471624] WARNING: CPU: 8 PID: 527 at lib/refcount.c:25 refcount_warn_saturate+0x74/0x110
[  420.471635] Modules linked in: ice(E+) snd_seq_dummy(E) snd_hrtimer(E) snd_seq(E) snd_seq_device(E) snd_timer(E) snd(E) soundcore(E) nft_fib_inet(E) nft_fib_ipv4(E) nft_fib_ipv6(E) nft_fib(E) nft_reject_inet(E) nf_reject_ipv4(E) nf_reject_ipv6(E) nft_reject(E) nft_ct(E) nft_chain_nat(E) nf_nat(E) nf_conntrack(E) nf_defrag_ipv6(E) nf_defrag_ipv4(E) rfkill(E) ip_set(E) nf_tables(E) nfnetlink(E) sunrpc(E) vfat(E) fat(E) intel_rapl_msr(E) intel_rapl_common(E) amd64_edac(E) ipmi_ssif(E) edac_mce_amd(E) kvm_amd(E) kvm(E) irqbypass(E) acpi_ipmi(E) ipmi_si(E) rapl(E) pcspkr(E) acpi_cpufreq(E) ipmi_devintf(E) i2c_piix4(E) k10temp(E) ipmi_msghandler(E) joydev(E) acpi_power_meter(E) xfs(E) libcrc32c(E) sd_mod(E) t10_pi(E) crc64_rocksoft_generic(E) crc64_rocksoft(E) crc64(E) sg(E) mgag200(E) crct10dif_pclmul(E) crc32_pclmul(E) crc32c_intel(E) drm_kms_helper(E) syscopyarea(E) sysfillrect(E) ghash_clmulni_intel(E) ahci(E) sysimgblt(E) i2c_algo_bit(E) libahci(E) drm_shmem_helper(E) sha512_ssse3(E) libata(E) drm(E) tg3(E) i40e(E)
[  420.471709]  ccp(E) sp5100_tco(E) megaraid_sas(E) dm_mirror(E) dm_region_hash(E) dm_log(E) dm_mod(E) fuse(E)
[  420.471722] CPU: 8 PID: 527 Comm: kworker/8:2 Kdump: loaded Tainted: G            E      6.4.0-rc6+ #64
[  420.471725] Hardware name: Dell Inc. PowerEdge R7525/0YHMCJ, BIOS 1.7.3 10/05/2020
[  420.471728] Workqueue: events work_for_cpu_fn
[  420.471735] RIP: 0010:refcount_warn_saturate+0x74/0x110
[  420.471738] Code: 01 01 e8 7f cd b1 ff 0f 0b e9 f8 6a 5e 00 80 3d be e7 33 01 00 75 cb 48 c7 c7 f0 17 48 9a c6 05 ae e7 33 01 01 e8 5c cd b1 ff <0f> 0b e9 d5 6a 5e 00 80 3d 9d e7 33 01 00 75 a8 48 c7 c7 c8 17 48
[  420.471740] RSP: 0018:ffffb704488e7c00 EFLAGS: 00010286
[  420.471742] RAX: 0000000000000000 RBX: ffffffffc1cfb8a0 RCX: 0000000000000000
[  420.471744] RDX: 0000000000000002 RSI: ffffffff9a46bae8 RDI: 00000000ffffffff
[  420.471746] RBP: ffff94f8c69df628 R08: 0000000000000000 R09: 00000000ffff7fff
[  420.471747] R10: ffffb704488e7aa0 R11: ffffffff9a9dc1c8 R12: ffff94f8c69df620
[  420.471748] R13: ffff94f8d6a05c00 R14: ffff94f8c69df640 R15: ffff94f90f00ed40
[  420.471750] FS:  0000000000000000(0000) GS:ffff95083f800000(0000) knlGS:0000000000000000
[  420.471751] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  420.471753] CR2: 00007fece9379820 CR3: 00000008d17aa000 CR4: 0000000000350ee0
[  420.471754] Call Trace:
[  420.471759]  <TASK>
[  420.471764]  ? __warn+0x80/0x130
[  420.471768]  ? refcount_warn_saturate+0x74/0x110
[  420.471771]  ? report_bug+0x1c3/0x1d0
[  420.471776]  ? prb_read_valid+0x17/0x20
[  420.471780]  ? handle_bug+0x3c/0x70
[  420.471785]  ? exc_invalid_op+0x14/0x70
[  420.471788]  ? asm_exc_invalid_op+0x16/0x20
[  420.471798]  ? refcount_warn_saturate+0x74/0x110
[  420.471801]  dpll_xa_ref_pin_add+0x286/0x2c0
[  420.471806]  __dpll_pin_register+0x27/0xb0
[  420.471809]  dpll_pin_register+0xa6/0xe0
[  420.471813]  ice_dpll_register_pins+0x4e/0xc0 [ice]
[  420.471869]  ice_dpll_init_direct_pins.part.0+0x2f/0xc0 [ice]
[  420.471901]  ? ice_dpll_get_pins.constprop.0+0x4e/0xc0 [ice]
[  420.471932]  ice_dpll_init_pins+0x264/0x460 [ice]
[  420.471962]  ice_dpll_init+0xfa/0x250 [ice]
[  420.471994]  ice_init_features+0x75/0x270 [ice]
[  420.472030]  ice_probe+0x28a/0x2c0 [ice]
[  420.472059]  local_pci_probe+0x46/0xa0
[  420.472065]  work_for_cpu_fn+0x16/0x20
[  420.472068]  process_one_work+0x1e5/0x3f0
[  420.472072]  ? __pfx_worker_thread+0x10/0x10
[  420.472074]  worker_thread+0x1c4/0x3a0
[  420.472076]  ? __pfx_worker_thread+0x10/0x10
[  420.472078]  kthread+0xf1/0x120
[  420.472082]  ? __pfx_kthread+0x10/0x10
[  420.472085]  ret_from_fork+0x2c/0x50
[  420.472092]  </TASK>
[  420.472093] ---[ end trace 0000000000000000 ]---
[  420.472141] ------------[ cut here ]------------